Actions/DetermineSecrets/DetermineSecrets.ps1

```diff
@@ -0,0 +1,107 @@
+        }
+    }
+}
+
+AddSecret -secret 'AZURE_CREDENTIALS' -useMapping
+foreach($secret in ($getSecrets.Split(',') | Select-Object -Unique)) {
+    switch ($secret) {
+        'TokenForPush' {
+            AddSecret -secret 'TokenForPush'
+            if ($useGhTokenWorkflowForPush -eq 'true') {
+                # If we are using the ghTokenWorkflow for commits, we need to get ghTokenWorkflow secret
+                AddSecret -secret 'ghTokenWorkflow' -useMapping
+            }
+            else {
+                AddSecret -secret 'github_token'
+            }
+        }
+        'GitSubmodulesToken' {
+            # If we are getting the gitSubModules token, we might need to get the github token as well
+            AddSecret -secret $secret -useMapping
+            AddSecret -secret 'github_token'
+        }
+        'AppDependencySecrets' {
+            # Loop through appDependencyProbingPaths and trustedNuGetFeeds and add secrets to the collection of secrets to get
+            $settingsCollection = @()
+            if ($settings.Keys -contains 'appDependencyProbingPaths') {
+                $settingsCollection += $settings.appDependencyProbingPaths
+            }
+            if ($settings.Keys -contains 'trustedNuGetFeeds') {
+                $settingsCollection += $settings.trustedNuGetFeeds
+            }
+            foreach($settingsItem in $settingsCollection) {
+                if ($settingsItem.PsObject.Properties.name -eq "AuthTokenSecret") {
+                    AddSecret -secret $settingsItem.authTokenSecret
+                }
+            }
+            # Look through installApps and installTestApps for secrets and add them to the collection of secrets to get
+            foreach($installSettingsKey in @('installApps','installTestApps')) {
+                if ($settings.Keys -contains $installSettingsKey) {
+                    $settings."$installSettingsKey" | ForEach-Object {
+                        # If any of the installApps URLs contains '${{SECRETNAME}}' we need to get the secret
+                        $pattern = '.*(\$\{\{\s*([^}]+?)\s*\}\}).*'
+                        if ($_ -match $pattern) {
+                            AddSecret -secret $matches[2]
+                        }
+                    }
+                }
+            }
+        }
+        default {
+            AddSecret -secret $secret -useMapping
+        }
+    }
+}
+
+# Calculate output for secrets
+# one output called FORMATSTR with the content: {{"secret1":{0},"secret2":{1},"secret3":{2}}}
+# and one environment variable per secret called S0, S1, S2 with the name of the GitHub Secret (or Azure DevOps secret) to look for
+if ($script:secretsCollection.Count -gt 32) {
+    throw "Maximum number of secrets exceeded."
+}
+
+$cnt = 0
+$formatArr = @()
+foreach($secret in $script:secretsCollection) {
+    $formatArr += @("""$Secret"":{$cnt}")
+    Add-Content -Encoding UTF8 -Path $ENV:GITHUB_ENV -Value "S$cnt=$($script:secretNames[$secret])"
```

Comment:
I think it's better to add these in the output as well.

The action does define S0, S1, etc. outputs.

Reply:
The action lists them as output because I had them there in the beginning.
I moved them to env in order to shorten the length of the FORMAT_JSON line - and to make sure that all instances of that line are equal.
If you want me to put them back as OUTPUT - I can do that - or I can remove the OUTPUT definitions.
(I think the latter makes most sense)

Reply:
The line would go from:

${{ format(steps.DetermineSecrets.outputs.formatStr,toJSON(secrets[env.S0]),toJSON(secrets[env.S1]),toJSON(secrets[env.S2]),toJSON(secrets[env.S3]),toJSON(secrets[env.S4]),toJSON(secrets[env...

to

```
${{ format(steps.DetermineSecrets.outputs.formatStr,toJSON(secrets[steps.DetermineSecrets.outputs.S0]),toJSON(secrets[steps.DetermineSecrets.outputs.S1]),toJSON(secrets[steps.DetermineSecrets.outputs.S2]),toJSON(secrets[steps.DetermineSecrets.outputs.S3]),toJSON(secrets[steps.DetermineSecrets.outputs.S4]),toJSON(secrets[steps.DetermineSecrets.outputs...
```

Neither is perfect, but I would prefer 1 over 2 - your call?

Reply:
2 is indeed longer, but 1 pollutes the env variables unnecessarily. Screenshot from https://github.com/freddydk/test/actions/runs/15132554080/job/42536747440

<img width="593" alt="image" src="https://github.com/user-attachments/assets/6e172107-e823-451d-9c13-b5afbb3133a1" />


People don't look at yaml as often as the look at the logs. So, my vote goes for 2 if there are no other limitations.

Reply:
2 it is - your wish is my command :-)